Content.Shared/Access/Systems/AccessReaderSystem.cs

```diff
@@ -100,17 +104,53 @@ public bool IsAllowed(EntityUid user, EntityUid target, AccessReaderComponent? r
         if (!reader.Enabled)
             return true;
 
+        var validIds = GetAllowedEntities(user, reader).ToList();
+        // User is in a box.
+        if (TryComp(user, out CardboardBoxComponent? cardboardBox) && cardboardBox.Mover.HasValue)
```

Comment:
You should probably use an event for this. Maybe something like `IsInInsideSomething` event that the `CardboardBoxComponent` will listen for and respond to with the person inside! This will make it a lot more reusable and less hard coded.

Reply:
Yeah, the borg chassis and cardboard box should both be handled by an event. The way you are doing it hardcodes both both components into the access reader. If a fork (for example RMC) wanted to remove one of them, they would have to completely rewrite your PR. Events are more modular.